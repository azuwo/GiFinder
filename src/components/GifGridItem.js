import React from 'react'

const GifGridItem = ({title,url}) => {
    return (
        <div className = 'card'>
            <img src={url} alt={title}/>
        </div>
    )
}

export default GifGridItem
