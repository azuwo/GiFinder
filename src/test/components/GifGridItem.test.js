import { shallow } from 'enzyme';
import React from 'react'
import GifGridItem from '../../components/GifGridItem';

describe('Test in GifGridItem', () => {

    const title = 'Hola'
    const url= 'https://someurl.com'
    const wrapper = shallow(<GifGridItem title={title} url={url}/>)

    test('should show component correctly', () => {

        expect(wrapper).toMatchSnapshot()
    })

/*     test('should have <p> with a title', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title);
    }); */

    test('should contains an image and alt by props', () => {
        const {src,alt} = wrapper.find('img').props()
        expect(src).toBe(url)
        expect(alt).toBe(title)

    });

    test('should contain class', () => {
        const {className} = wrapper.find('div').props()
        expect(className.includes('card')).toBe(true)
    });
})
