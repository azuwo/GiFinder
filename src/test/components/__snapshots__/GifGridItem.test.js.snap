// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifGridItem should show component correctly 1`] = `
<div
  className="card"
>
  <img
    alt="Hola"
    src="https://someurl.com"
  />
</div>
`;
